gadgets/trace_tcpretrans/test/trace_tcpretrans_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
+	igtesting "github.com/inspektor-gadget/inspektor-gadget/pkg/testing"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/containers"
+	igrunner "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/ig"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/match"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/utils"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type traceTCPretransEvent struct {
+	eventtypes.CommonData
+
+	Timestamp string `json:"timestamp"`
+	MntNsID   uint64 `json:"mntns_id"`
+	NetNs     uint64 `json:"netns_id"`
+
+	Pid uint32 `json:"pid"`
+	Tid uint32 `json:"tid"`
+	Uid uint32 `json:"uid"`
+	Gid uint32 `json:"gid"`
+
+	Comm string           `json:"comm"`
+	Type string           `json:"type"`
+	Dst  utils.L4Endpoint `json:"dst"`
```

Comment:
Why is src missing?

Reply:
Hi @mauriciovasquezbernal,
I wasn't sure about the IP for the container, and I also wanted to ask about using tc to emulate packet drop, which requires NET_ADMIN capabilities. With my understanding of IG, there isn't a way to add individual capabilities to test containers. As a result, I am running the container with privileges, which may cause errors with K8s since it doesn't support privileged containers. Is there a workaround for this? Should we add something like container.AddCapabilities() to testutils? I hope I didn't miss something obvious.
Thanks!

Reply:
I think we can at least check that the field is present. (Using the normalize logic)

Regarding capabilities and privileged, I think for testing purposes it's just fine to be able to run the pod / container as privileged. Regarding the k8s thing, I'd suggest to skip this test on k8s, you can use something like:

https://github.com/inspektor-gadget/inspektor-gadget/blob/84d2519362b494848d31d4ddf9cfa2fcbf87cf37/gadgets/trace_dns/test/trace_dns_test.go#L71-L73

(checking for utils.CurrentTestComponent == utils.KubectlGadgetTestComponent should work)

Reply:
@mauriciovasquezbernal I have made the desired changes and squashed the commits for this one, Thanks!